backend/store/postgres/watcher.go

```diff
@@ -49,3 +63,139 @@ func (rs recordStatus) Row(id string, resource storev2.Wrapper) poll.Row {
 	}
 	return row
 }
+
+type Watcher struct {
+	store          WatchedStore
+	watchInterval  time.Duration
+	watchTxnWindow time.Duration
+}
+
+func NewWatcher(store WatchedStore, watchInterval time.Duration, watchTxnWindow time.Duration) *Watcher {
+	return &Watcher{
+		store:          store,
+		watchInterval:  watchInterval,
+		watchTxnWindow: watchTxnWindow,
+	}
+}
+
+func (w *Watcher) WatchConfig(ctx context.Context, req storev2.ResourceRequest) <-chan []storev2.WatchEvent {
+	return w.watch(ctx, req, true)
+}
+
+func (w *Watcher) Watch(ctx context.Context, req storev2.ResourceRequest) <-chan []storev2.WatchEvent {
+	return w.watch(ctx, req, false)
+}
+
+func (w *Watcher) watch(ctx context.Context, req storev2.ResourceRequest, isConfig bool) <-chan []storev2.WatchEvent {
+	eventChan := make(chan []storev2.WatchEvent, 32)
+
+	var table poll.Table
+
+	tableFactory, ok := getWatchStoreOverride(req.StoreName)
+	if !ok {
+		if isConfig {
+			tableFactory = watchConfigStoreOverride
+		} else {
+			tableFactory = func(storev2.ResourceRequest, *pgxpool.Pool) (poll.Table, error) {
+				return nil, fmt.Errorf("default watcher not yet implemented")
+			}
+		}
+	}
+	table, err := tableFactory(req, w.store.GetPgxPool())
```

Comment:
Now that we have separated storev2.Interface implementations into ConfigStore and (roughly) a state store do we still need this plumbing?

Doesn't need to happen now, but perhaps the WatchedStore interface could be expanded to replace most of this?